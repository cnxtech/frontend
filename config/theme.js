import theme from '@emcasa/ui'
import {fas} from '@fortawesome/pro-solid-svg-icons'
import {fal} from '@fortawesome/pro-light-svg-icons'
import icons from './icons'

export default {
  ...theme,
  icons: {emcasa: icons, default: fas, light: fal}
}
