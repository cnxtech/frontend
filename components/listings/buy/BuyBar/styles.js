import styled from 'styled-components'
import theme from '@emcasa/ui'
import Row from '@emcasa/ui-dom/components/Row'
import {LISTING_MAX_WIDTH} from 'constants/dimensions'

export const Wrapper = styled.div`
  width: 100%;
  background: ${theme.colors.white};
  border-top: 1px solid ${theme.colors.smoke};
`

export const Container = styled(Row)`
  justify-content: space-between;
  width: 100%;
  max-width: ${LISTING_MAX_WIDTH}px;
  padding: ${theme.space[2]}px ${theme.space[4]}px;
  margin: auto;
  box-sizing: border-box;
`

export const FavCount = styled(Row)`
  align-items: center;

  svg {
    width: 15px;

    path {
      fill: ${theme.colors.white};
      stroke: ${theme.colors.pink};
      stroke-width: 40;
    }
  }

  p {
    display: flex;
    align-items: center;
    margin: 0 0 0 ${theme.space[3]}px;
  }

  span {
    margin-left: ${theme.space[1]}px;
  }
`
