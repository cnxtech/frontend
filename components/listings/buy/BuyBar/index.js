import {Component, Fragment} from 'react'
import theme from '@emcasa/ui'
import Text from '@emcasa/ui-dom/components/Text'
import FontAwesomeIcon from '@fortawesome/react-fontawesome'
import faHeart from '@fortawesome/fontawesome-free-solid/faHeart'
import faSearch from '@fortawesome/fontawesome-free-solid/faSearch'
import Button from 'components/shared/Common/Buttons'
import {Wrapper, Container, FavCount} from './styles'

class BuyBar extends Component {
  render() {
    const {user} = this.props
    return (
      <Wrapper>
        <Container>
          <FavCount>
            {user.authenticated && (
              <Fragment>
                <FontAwesomeIcon icon={faHeart} size="1x" />
                <Text fontSize={1}>Favoritos<Text fontSize={1} inline color="pink">4</Text></Text>
              </Fragment>
            )}
          </FavCount>
          <Button fontSize={1} noBorder iconColor={theme.colors.pink}>
            <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 21 16">
              <g fill="none" fillRule="evenodd">
                <path d="M-2-4h24v24H-2z"/>
                <path
                  stroke={theme.colors.pink}
                  strokeLinecap="round"
                  strokeWidth="2"
                  d="M19.5 3.5H6h13.5zm-4.5 9H1.10546875 15zM3.5 6C2.11928813 6 1 4.88071187 1 3.5S2.11928813 1 3.5 1 6 2.11928813 6 3.5 4.88071187 6 3.5 6zm14 9c-1.3807119 0-2.5-1.1192881-2.5-2.5s1.1192881-2.5 2.5-2.5 2.5 1.1192881 2.5 2.5-1.1192881 2.5-2.5 2.5z"
                />
              </g>
            </svg>
            Filtro
          </Button>
        </Container>
      </Wrapper>
    )
  }
}

export default BuyBar
