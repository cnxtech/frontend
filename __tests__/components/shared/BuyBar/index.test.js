import React from 'react'
import BuyBar from 'components/shared/BuyBar'
import renderer from 'react-test-renderer'
import 'jest-styled-components'
import theme from 'config/theme'
import {ThemeProvider} from 'styled-components'
import {MockedProvider} from 'react-apollo/test-utils'

describe('<BuyBar/>', () => {
  it('should render component', () => {
    const tree = renderer.create(
      <ThemeProvider theme={theme}>
        <MockedProvider>
          <BuyBar
            user={{authenticated: false}}
          />
        </MockedProvider>
      </ThemeProvider>).toJSON()
    expect(tree).toMatchSnapshot()
  })
})
