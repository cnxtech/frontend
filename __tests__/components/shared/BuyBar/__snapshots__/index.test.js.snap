// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<BuyBar/> should render component 1`] = `
.c3 {
  outline: none;
  cursor: pointer;
  padding: 0 20px 0 20px;
  border-color: #e0e6ed;
  background-color: #ffffff;
  border-radius: 4px;
  border-width: 1px;
  width: auto;
  overflow: hidden;
  text-overflow: ellipsis;
  white-space: nowrap;
  height: 40px;
  color: #3c4858;
  font-family: Rubik;
  font-size: 14px;
  -webkit-font-smoothing: antialiased;
  border-style: solid;
  border: none;
}

.c3 svg {
  color: #ff2b7f;
  fill: #ff2b7f;
  margin-right: 10px;
  display: inline-block;
  width: 1em;
  height: 1em;
  overflow: visible;
  vertical-align: -0.125em;
  font-size: inherit;
}

.c0 {
  width: 100%;
  background: #ffffff;
  border-top: 1px solid #E0E6ED;
}

.c1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  width: 100%;
  max-width: 1224px;
  padding: 10px 20px;
  margin: auto;
  box-sizing: border-box;
}

.c2 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c2 svg {
  width: 15px;
}

.c2 svg path {
  fill: #ffffff;
  stroke: #ff2b7f;
  stroke-width: 40;
}

.c2 p {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  margin: 0 0 0 14px;
}

.c2 span {
  margin-left: 4px;
}

<div
  className="c0"
>
  <div
    className="c1"
  >
    <div
      className="c2"
    />
    <button
      className="c3"
      fontSize={1}
      height="medium"
    >
      <svg
        viewBox="0 0 21 16"
        xmlns="http://www.w3.org/2000/svg"
      >
        <g
          fill="none"
          fillRule="evenodd"
        >
          <path
            d="M-2-4h24v24H-2z"
          />
          <path
            d="M19.5 3.5H6h13.5zm-4.5 9H1.10546875 15zM3.5 6C2.11928813 6 1 4.88071187 1 3.5S2.11928813 1 3.5 1 6 2.11928813 6 3.5 4.88071187 6 3.5 6zm14 9c-1.3807119 0-2.5-1.1192881-2.5-2.5s1.1192881-2.5 2.5-2.5 2.5 1.1192881 2.5 2.5-1.1192881 2.5-2.5 2.5z"
            stroke="#ff2b7f"
            strokeLinecap="round"
            strokeWidth="2"
          />
        </g>
      </svg>
      Filtro
    </button>
  </div>
</div>
`;
